homeassistant/helpers/script.py

```diff
@@ -166,12 +196,31 @@ def _async_check_condition(self, action, variables):
         self._log("Test condition {}: {}".format(self.last_action, check))
         return check
 
+    def _async_set_timeout(self, action, variables):
+        """Schedule a timeout to abort script."""
+        timeout = action[CONF_TIMEOUT]
+
+        @callback
+        def async_script_timeout(now):
+            """Call after timeout is retrieve stop script."""
+            self._async_unsub_delay_listener = None
+            self.async_stop()
+
+        self._async_unsub_delay_listener = \
```

Comment:
You're reusing a listener variable here, that is confusing. I think that we should rewrite it to keep track of an array of listeners. Unsub can then become:

```python
def _async_remove_listeners(self):
    for unsub in self._unsub_listeners:
        unsub()
    self._unsub_listeners = []
```